x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTimeFunction.java

```diff
@@ -33,12 +36,17 @@ protected Object doFold(ZonedDateTime dateTime) {
         return dateTimeChrono(dateTime, extractor.chronoField());
     }
 
+    public static Integer dateTimeChrono(OffsetTime time, String tzId, String chronoName) {
+        return dateTimeChrono(time.withOffsetSameInstant(ZoneId.of(tzId).getRules().getOffset(Instant.now())),
+            ChronoField.valueOf(chronoName));
+    }
+
     public static Integer dateTimeChrono(ZonedDateTime dateTime, String tzId, String chronoName) {
         ZonedDateTime zdt = dateTime.withZoneSameInstant(ZoneId.of(tzId));
         return dateTimeChrono(zdt, ChronoField.valueOf(chronoName));
     }
 
-    private static Integer dateTimeChrono(ZonedDateTime dateTime, ChronoField field) {
+    private static Integer dateTimeChrono(Temporal dateTime, ChronoField field) {
```

Comment:
Isn't `Temporal` too broad here?

Reply:
It's an internal method, called only with ZonedDateTime or OffsetTime so it's not dangerous and avoids duplication.